### ISSUE TITLE
Can't encode bitdepth lower than 8?

### ISSUE THREAD

**Issue by lunasorcery** [kornelski/lodepng-rust#60]
Unless I've missed something, it seems that attempts to encode an image of 1/2/4-bitdepth will fail.
`buffer_for_type` checks the data size as such:
```rs
let bytes_per_pixel = bitdepth as usize/8;
assert!(mem::size_of::<PixelType>() <= 4*bytes_per_pixel, "Implausibly large {}-byte pixel data type", mem::size_of::<PixelType>());
```
For an input bitdepth of less than 8, `bytes_per_pixel` will be 0, and then the assert cannot possibly pass unless `PixelType` is somehow a zero-size type.

**Comment by kornelski** [comment]
That's correct. In the RGBA mode PNG supports only 8 and 16 bit depths. Lower bits per pixel are possible only in palette modes.

**Comment by lunasorcery** [comment]
In this case I was trying to encode in a non-paletted Grey mode, which should support 1,2,4,8,16 bit.

**Comment by kornelski** [comment]
Oh, you're right. I've reworked that check, and released a fix in 3.9.2.
